docs/contribute.rst

```diff
@@ -325,6 +325,56 @@ Multiple submodules
 
 The modules system is hierachical and one module may contain multiple submodules. For example uniswap having both v1 and v3 each in their own subdirectories as seen `here <https://github.com/rotki/rotki/tree/develop/rotkehlchen/chain/ethereum/modules/uniswap>`__.
 
+Add new language or translation
+===================================
+
+Add new language
+----------------
+The translation files is located `here <https://github.com/rotki/rotki/tree/develop/frontend/app/src/locales>`__.
+They are saved with format ``{country_code}.json``, where country code is ``Alpha-2`` code from based on `ISO 3166 country codes <https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes>`__.
+If you want to add new language, you need to create new translation file with that format, and then `fill it <#add-or-edit-translation>`__.
+
+You also need to update the frontend mapping that defined at this `enum <https://github.com/rotki/rotki/blob/f57522baa737854e6affcbe57bada2b81c4dee83/frontend/app/src/types/frontend-settings.ts#L112>`__, and this `entries <https://github.com/rotki/rotki/blob/f57522baa737854e6affcbe57bada2b81c4dee83/frontend/app/src/data/supported-language.ts>`__.
+
+Add or edit translation
+-----------------------
+Rotki do translation using `Vue i18n <https://kazupon.github.io/vue-i18n>`__.
+
+Rotki main language is ``English``. The translation file is `here <https://github.com/rotki/rotki/blob/develop/frontend/app/src/locales/en.json>`__.
+In order to fill the translation for other language, you should pay attention to these few things:
+
+1. The `JSON` structure from `English` translation file is absolute, meaning you can't change the JSON structure (the keys), because this is how rotki reads which value to use. So for translation on other languages, please follow the same structure from `English` translation JSON. For example:
```

Comment:
```suggestion
1. The ``JSON`` structure from the ``English`` language file is absolute, meaning you can't change the JSON structure (the keys), because this is how rotki reads which value to use. So for translations of other languages, please follow the same structure as the `English` language JSON file. For example:
```